third_party/move/move-compiler-v2/src/pipeline/livevar_analysis_processor.rs

```diff
@@ -93,17 +118,52 @@ impl LiveVarInfoAtCodeOffset {
     }
 }
 
+/// Usages of a given temporary after this program point.
+/// There is at least one usage.
 #[derive(Debug, Clone, Eq, PartialEq, PartialOrd)]
-pub struct LiveVarInfo {
-    /// The usage of a given temporary after this program point, inclusive of locations where
-    /// the usage happens. This set contains at least one element.
-    pub usages: BTreeSet<Loc>,
+pub enum LiveVarInfo {
+    /// Location where the temporary is used. Only unshadowed uses are tracked.
+    UsageLocs(OrdSet<Loc>),
```

Comment:
This is surprising. If you need code offsets in addition, why not just add them instead of making this an enum which all code now needs to switch over?

Reply:
Changed, based on our offline discussion.